### ISSUE TITLE
ShardDomain results in unoptimized CSS

### ISSUE THREAD

**Issue by injust** [apache/incubator-pagespeed-ngx#1321]
When ShardDomain directive is used to relocate resources to a second domain (for static resources only), CSS served through PageSpeed at the new (and old) URL shows some odd behaviour.
The URL is rewritten into a .pagespeed. URL and multiple CSS files are concatenated together, but the code itself is not minifed.

Response headers are also different:

Without ShardDomain
```
cache-control:max-age=31536000
content-encoding:gzip
content-type:text/css
date:Mon, 14 Nov 2016 03:54:57 GMT
etag:W/"0"
expires:Tue, 14 Nov 2017 03:54:57 GMT
last-modified:Mon, 14 Nov 2016 03:54:57 GMT
server:nginx/1.11.5
status:200
vary:Accept-Encoding
vary:Accept-Encoding
x-original-content-length:93603
x-page-speed:1.11.33.4-0
```

With ShardDomain
```
cache-control:max-age=300,private
content-encoding:gzip
content-type:text/css
date:Mon, 14 Nov 2016 03:55:31 GMT
expires:Mon, 14 Nov 2016 04:00:31 GMT
last-modified:Mon, 14 Nov 2016 03:55:31 GMT
server:nginx/1.11.5
status:200
vary:Accept-Encoding
vary:Accept-Encoding
x-content-type-options:nosniff
x-original-content-length:104675
x-page-speed:1.11.33.4-0
```

**Comment by jmarantz** [comment]
could you post your configuration and an example URL?

Also, appending "?PageSpeedFilters=+debug" will cause the HTML output to be annotated with debug messages, which may help.

**Comment by Injust** [comment]
nginx configuration
```
error_log /var/log/nginx/error.log;
pid /var/run/nginx.pid;
user www-data;

events {}

http {
	access_log /var/log/nginx/access.log;
	default_type application/octet-stream;
	include /etc/nginx/mime.types;
	index index.html index.php;
	ssl_certificate /etc/nginx/ecdsa/fullchain.pem;
	ssl_certificate_key /etc/nginx/ecdsa/privkey.pem;

	pagespeed FileCachePath /var/cache/pagespeed;
	#pagespeed ModifyCachingHeaders off;
	#pagespeed ShardDomain https://asmui.ml https://asmui.ga;
	pagespeed on;

	server {
		listen 443 ssl spdy http2;
		listen [::]:443 ssl spdy http2;
		root /var/www/asm;
		server_name asmui.ga asmui.ml;

		location ~ "\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+" {
			add_header "" "";
		}
		location ~ ^/pagespeed_static/ {}
		location ~ ^/ngx_pagespeed_beacon$ {}

		location ~ \.php$ {
			include /etc/nginx/fastcgi.conf;
		}
	}
}
```

Without ShardDomain
```
mod_pagespeed on
Filters:
ah	Add Head
cc	Combine Css
jc	Combine Javascript
gp	Convert Gif to Png
jp	Convert Jpeg to Progressive
jw	Convert Jpeg To Webp
mc	Convert Meta Tags
pj	Convert Png to Jpeg
ws	When converting images to WebP, prefer lossless conversions
db	Debug
ec	Cache Extend Css
ei	Cache Extend Images
es	Cache Extend Scripts
fc	Fallback Rewrite Css 
if	Flatten CSS Imports
hw	Flushes html
ci	Inline Css
ii	Inline Images
il	Inline @import to Link
ji	Inline Javascript
js	Jpeg Subsampling
rj	Recompress Jpeg
rp	Recompress Png
rw	Recompress Webp
ri	Resize Images
cf	Rewrite Css
jm	Rewrite External Javascript
jj	Rewrite Inline Javascript
cu	Rewrite Style Attributes With Url
cp	Strip Image Color Profiles
md	Strip Image Meta Data

Options:
AvoidRenamingIntrospectiveJavascript (aris) True
EnableRewriting (e) 1
FileCacheInodeLimit (afcl) 500000
RewriteLevel (l) Core Filters

#NumFlushes            0
#EndDocument after     888us
#Total Parse duration  791us
#Total Render duration 1032us
#Total Idle duration   97us
No critical images detected.
The following filters were disabled for this request:
	SupportNoscript
```
```
cache-control:max-age=31536000
content-encoding:gzip
content-type:text/css
date:Mon, 14 Nov 2016 15:29:23 GMT
etag:W/"0"
expires:Tue, 14 Nov 2017 15:29:23 GMT
last-modified:Mon, 14 Nov 2016 15:29:23 GMT
server:nginx/1.11.5
status:200
vary:Accept-Encoding
vary:Accept-Encoding
x-original-content-length:93603
x-page-speed:1.11.33.4-0
```
HTML source gives:
```
<link rel="stylesheet" href="/res/css/A.fontello.css+paper.css+scrollbar.css+style.css,Mcc.pyRZM-96yH.css.pagespeed.cf.yyQUpjEnFC.css"><!--CSS not inlined since it&#39;s bigger than 2048 bytes--><!--Uncacheable content, preventing rewriting of https://asmui.ml/res/img/logo.svg--><!--Uncacheable content, preventing rewriting of https://asmui.ml/res/img/logo.svg-->
```

With ShardDomain
```
mod_pagespeed on
Filters:
ah	Add Head
cc	Combine Css
jc	Combine Javascript
gp	Convert Gif to Png
jp	Convert Jpeg to Progressive
jw	Convert Jpeg To Webp
mc	Convert Meta Tags
pj	Convert Png to Jpeg
ws	When converting images to WebP, prefer lossless conversions
db	Debug
ec	Cache Extend Css
ei	Cache Extend Images
es	Cache Extend Scripts
fc	Fallback Rewrite Css 
if	Flatten CSS Imports
hw	Flushes html
ci	Inline Css
ii	Inline Images
il	Inline @import to Link
ji	Inline Javascript
js	Jpeg Subsampling
rj	Recompress Jpeg
rp	Recompress Png
rw	Recompress Webp
ri	Resize Images
cf	Rewrite Css
jm	Rewrite External Javascript
jj	Rewrite Inline Javascript
cu	Rewrite Style Attributes With Url
cp	Strip Image Color Profiles
md	Strip Image Meta Data

Options:
AvoidRenamingIntrospectiveJavascript (aris) True
EnableRewriting (e) 1
FileCacheInodeLimit (afcl) 500000
RewriteLevel (l) Core Filters

#NumFlushes            0
#EndDocument after     766us
#Total Parse duration  601us
#Total Render duration 1511us
#Total Idle duration   165us
No critical images detected.
The following filters were disabled for this request:
	SupportNoscript
```
```
cache-control:max-age=300,private
content-encoding:gzip
content-type:text/css
date:Mon, 14 Nov 2016 15:35:28 GMT
expires:Mon, 14 Nov 2016 15:40:28 GMT
last-modified:Mon, 14 Nov 2016 15:35:28 GMT
server:nginx/1.11.5
status:200
vary:Accept-Encoding
vary:Accept-Encoding
x-content-type-options:nosniff
x-original-content-length:104675
x-page-speed:1.11.33.4-0
```
HTML source gives:
```
<link rel="stylesheet" href="https://asmui.ga/res/css/A.fontello.css+paper.css+scrollbar.css+style.css,Mcc.ABVRo4Hhqo.css.pagespeed.cf.ujtxA5uFVz.css"><!--CSS not inlined since it&#39;s bigger than 2048 bytes--><!--Uncacheable content, preventing rewriting of https://asmui.ga/res/img/logo.svg--><!--Uncacheable content, preventing rewriting of https://asmui.ga/res/img/logo.svg-->
```

Also to note, https://asmui.ml/res/css/A.fontello.css+paper.css+scrollbar.css+style.css,Mcc.pyRZM-96yH.css.pagespeed.cf.yyQUpjEnFC.css is minified, but https://asmui.ga/res/css/A.fontello.css+paper.css+scrollbar.css+style.css,Mcc.ABVRo4Hhqo.css.pagespeed.cf.ujtxA5uFVz.css is not.

**Comment by injust** [comment]
Closing due to age.
Reportedly fixed in version `1.12`, according to apache/incubator-pagespeed-mod#1659.
